tree/dataframe/src/RLoopManager.cxx

```diff
@@ -501,12 +533,13 @@ void RLoopManager::RunTreeProcessorMT()
 /// Run event loop over one or multiple ROOT files, in sequence.
 void RLoopManager::RunTreeReader()
 {
+   if (fEndEntry == fStartEntry) // empty range => no work needed
+      return;
+
    TTreeReader r(fTree.get(), fTree->GetEntryList());
 
-   if (fEndEntry > fStartEntry) {
-      // User provided a valid entry range for the tree
-      r.SetEntriesRange(fStartEntry, fEndEntry);
-   }
+   if (r.SetEntriesRange(fStartEntry, fEndEntry)) // returning > 0 indicates invalid fStartEntry
+      throw std::runtime_error("RLoopManager: fStartEntry cannot be larger than the number of entries.");
```

Comment:
Are we sure this is the only case of invalid start entry for TTreeReader?

Reply:
afaik yes. Because in case `fStartEntry > fEndEntry` I would have errored out in ctor. Moreover, if the range is `fStartEntry < "actual last entry" < fEndEntry` then `SetEntriesRange` returns 0 and will output the entries starting from the `fStartEntry` up to the the actual last entry.

Reply:
doublechecking https://github.com/root-project/root/blob/a402af862ace9573193500bcccaa8b4d56cc6632/tree/treeplayer/src/TTreeReader.cxx#L411 this looks like the only invalid thing that can go wrong at this stage.